File: frontend-angular/src/app/models/personne-impliquee.ts
import { Affaire } from './affaire';

export interface PersonneImpliquee {
  id?: number;
  nom: string;
  prenom: string;
  dateNaissance: string;
  genre: string;
  telephone: string;
  adresse: string;
  typePersonne: string;
  notes: string;
  affaire?: Affaire;
}

File: backend-springboot/src/main/java/com/crimeai/backend_springboot/entity/PersonneImpliquee.java
package com.crimeai.backend_springboot.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "personnes_impliquees")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PersonneImpliquee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;

    private String prenom;

    private LocalDate dateNaissance;

    private String genre;

    private String telephone;

    private String adresse;

    private String typePersonne;
    // SUSPECT, VICTIME, TEMOIN

    @Column(columnDefinition = "TEXT")
    private String notes;

    @ManyToOne
    @JoinColumn(name = "affaire_id")
    private Affaire affaire;
}